Use Jekyll for templating

benbalter opened the pull request:
This pull request moves the site from the node-based custom build system to Jekyll, which is baked directly into GitHub Pages. The big benefit here (beyond not reinventing the wheel), is that you no longer have to commit the compiled site to the repository, thus pull requests to markdown instantly update the site, with no technical knowledge or need to run locally required.  Specifically:
- Move chapters from `/md` to `_chapters`
- Move templates to `_layouts` (note: I kept them as two different templates, but believe in practice they are identical)
- DRY things up by moving header/footer to partials (`_includes`) Note: they were a bit out of sync
- Use YAML frontmatter to store document metadata (e.g., title)
- Dynamically generate table of contents (summary) since chapters are now machine-readable
- Use bower to version vendored assets (bootstrap, jquery, modernizr)
- Use Pygments for syntax highlighting
- Minor markdown changes to better adhere to the spec (e.g., wrapping special characters in code blocks, colons before footnote definitions)
- Add sitemap for better SEO

The output should remain nearly identical, and there should be no substantive changes to the content. Once merged, potential contributors can propose changes to content by simply modifying the files int he `_chapters` folder, with no need to run locally or risk that they edit the output, etc.

benbalter committed:
initial jekyll commit

benbalter committed:
remove bower bloat

benbalter committed:
note how to run locally and bootstrap

benbalter committed:
few code blocks snuck by

benbalter committed:
one last code block

benbalter committed:
whitespace

benbalter committed:
fix faq link

benbalter committed:
Merge branch 'gh-pages' into jekyllification

Conflicts:
	_chapters/ch6.md

benbalter committed:
fix edit link

benbalter commented:
You can see the resulting folder structure over in https://github.com/benbalter/codingforlawyers/tree/jekyllification.

konklone commented:
Seems like a good move, but I suggest cutting out some of the root directory cruft and making the app more approachable to newcomers and potentially novice coders (which this project may disproportionately attract).

The site's not doing anything fancy, so I think the `github-pages` gem dependency can go, and `bower` seems like overkill. So I think these files the PR adds could be cut:
- `.bowerrc`
- `Gemfile`
- `Gemfile.lock`
- `bower.json`

I also think the Sass conversion could be undone, so that novices to HTML and CSS have an easier time chipping in.

These weren't added by this PR, but while I'm suggesting things to remove from the root -
- move `favicon.ico` and `apple-touch-icon-precomposed.png` into the `assets/` dir
- make a `pages/` dir and move `index.md` and `faq.md` into it
- ditch `404.html`

benbalter committed:
remove technical cruft

benbalter committed:
remove boilerplate

benbalter committed:
move pages to /pages

benbalter committed:
update usage instructions

benbalter committed:
git ignore gemfile so things actually work

benbalter committed:
ignore gemfile.lock :gem:

benbalter committed:
use _pages for pages for consistency

benbalter committed:
:skull: partials

benbalter committed:
dat css :lipstick:

benbalter commented:
>  making the app more approachable to newcomers and potentially novice coders (which this project may disproportionately attract).

Great point. Really like that idea. Also nuked the header/footer partials under the same rationale.

> The site's not doing anything fancy, so I think the github-pages gem dependency can go, and bower seems like overkill.

:+1:. Added Gemfile to gitignore, and can still user bower to update stuffs via cli.

> These weren't added by this PR, but while I'm suggesting things to remove from the root

:+1: Moved pages to `_pages` for consistency
